crates/aggchain-proof-core/src/full_execution_proof.rs

```diff
@@ -14,6 +14,37 @@ use crate::{error::ProofError, vkey_hash::HashU32};
 /// Hardcoded for now, might see if we might need it as input
 pub const OUTPUT_ROOT_VERSION: [u8; 32] = [0u8; 32];
 
+/// L2PreRoot is the representation of the previous OutputRoot
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct L2PreRoot(pub Digest);
+
+impl From<L2PreRoot> for FixedBytes<32> {
+    fn from(value: L2PreRoot) -> FixedBytes<32> {
+        value.0 .0.into()
+    }
+}
+
+/// OutputRoot is the concatenation of the OutputRoot version + payload
+///
+/// Payload composed of `state_root`, `withdrawal_storage_root`,
+/// `latest_block_hash`
+///
+/// Details: https://specs.optimism.io/protocol/proposals.html#l2-output-commitment-construction
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct OutputRoot(pub Digest);
```

Comment:
Nit: Some conversion methods like `to_array`, `to_fixed_bytes`, etc. with explicit names would be nice to avoid the `.0 .0` field accesses.